manage-data/data-store/data-streams/failure-store.md

```diff
@@ -0,0 +1,611 @@
+      },
+      "indices": [
+        {
+          "index_name": ".ds-my-datastream-2025.05.01-000001",
+          "index_uuid": "jUbUNf-8Re-Nca8vJkHnkA",
+          "managed_by": "Data stream lifecycle",
+          "prefer_ilm": true,
+          "index_mode": "standard"
+        }
+      ],
+      "generation": 2,
+      "status": "GREEN",
+      "template": "my-datastream-template",
+      "lifecycle": {
+        "enabled": true
+      },
+      "next_generation_managed_by": "Data stream lifecycle",
+      "prefer_ilm": true,
+      "hidden": false,
+      "system": false,
+      "allow_custom_routing": false,
+      "replicated": false,
+      "rollover_on_write": false,
+      "index_mode": "standard",
+      "failure_store": { <1>
+        "enabled": true,
+        "rollover_on_write": false,
+        "indices": [
+          {
+            "index_name": ".fs-my-datastream-2025.05.05-000002",
+            "index_uuid": "oYS2WsjkSKmdazWuS4RP9Q",
+            "managed_by": "Data stream lifecycle"  <2>
+          }
+        ],
+        "lifecycle": {
+          "enabled": true,
+          "effective_retention": "30d",  <3> 
+          "retention_determined_by": "default_failures_retention"  <4>
+        }
+      }
+    }
+  ]
+}
+```
+1. Information about the failure store is presented in the response under its own field.
+2. Indices are managed by data stream lifecycles by default.
+3. An effective retention period of thirty days (30d) is present by default.
+4. The retention is currently determined by the default.  
+
+:::{note}
+The default retention respects any maximum retention values. If [maximum retention](./failure-store.md) is configured lower than thirty days then the maximum retention will be used as the default value.
+:::
+
+You can update the default retention period for failure stores in your deployment by updating the `data_streams.lifecycle.retention.failures_default` cluster setting. Data streams that have no retention configured on their failure stores will use this value to determine their retention period.
+
+```console
+PUT _cluster/settings
+{
+  "persistent": {
+    "data_streams.lifecycle.retention.failures_default": "15d"
+  }
+}
+```
+
+You can also specify the failure store retention period for a data stream on its data stream options. These can be specified via the index template for new data streams, or via the [put data stream options](./failure-store.md) API for existing data streams.
+
+```console
+PUT _data_stream/my-datastream/_options
+{
+    "failure_store": {
+        "enabled": true, <1>
+        "lifecycle": {
+            "data_retention": "10d" <2>
+        }
+    }
+}
+```
+1. Ensure that the failure store remains enabled.
+2. Set only this data stream's failure store retention to ten days.
+
+### Add and remove from failure store [manage-failure-store-indices]
+
+Failure stores support adding and removing indices from them using the [modify data stream](./failure-store.md) API.
```

Comment:
What happens if you add a failure store backing index that has incompatible mappings? Are we doing validations when adding a backing index or would it fail at runtime? Not suggesting one way is better than the other but maybe we should describe what happens here to set expectations.

Reply:
There's no special handling for mappings when adding an index to the failure store. You could add a completely unrelated index to the failure store and we allow it. Indices that are added to a data stream can never be treated as a write index, so we're less worried about their mappings than when doing a rollover operation. Even if the failure store is empty and we add a random index, the failure store is still marked for lazy rollover and will create a write index on redirection.